datafusion/core/src/execution/context.rs

```diff
@@ -1810,22 +1810,14 @@ impl FunctionRegistry for SessionState {
     }
 }
 
-/// Task Context Properties
-pub enum TaskProperties {
-    ///SessionConfig
-    SessionConfig(SessionConfig),
```

Comment:
The core of the change is to just use `SessionConfig` everywhere, creating it in the constructor of `TaskContext` if needed